### PR TITLE
add task for Holocene System Config upgrade across `sepolia/{op,mode,metal,zora}`

### PR DESCRIPTION
**Pull request by geoknee**
Closes https://github.com/ethereum-optimism/optimism/issues/13029 

Note: I labelled this task 20 because of this prior task (not yet merged) #374

### CODE REVIEW

File: tasks/sep/020-holocene-system-config-upgrade-multi-chain/OVERVIEW.md
Comment (geoknee) [inline tasks/sep/020-holocene-system-config-upgrade-multi-chain/OVERVIEW.md:70]: I think we need to treat Base Sepolia differently?

Comment (mds1) [reply tasks/sep/020-holocene-system-config-upgrade-multi-chain/OVERVIEW.md]: Correct, they have a different owner on Sepolia so will need to be a different playbook

File: tasks/sep/020-holocene-system-config-upgrade-multi-chain/OVERVIEW.md
Comment (mds1) [inline tasks/sep/020-holocene-system-config-upgrade-multi-chain/OVERVIEW.md:1]: I'd delete this file for simplicity—it's not required, and it's just another thing that could get out of date when making changes to this PR, or the equivalent mainnet PR. The `input.json` and resulting `VALIDATION.md` are the source of truth

File: tasks/sep/020-holocene-system-config-upgrade-multi-chain/README.md
Comment (mds1) [inline tasks/sep/020-holocene-system-config-upgrade-multi-chain/README.md:3]: For sepolia, since we don't need foundation approval, we can skip directly to READY TO SIGN 
```suggestion
Status: READY TO SIGN
```

File: tasks/sep/020-holocene-system-config-upgrade-multi-chain/README.md
Comment (mds1) [inline tasks/sep/020-holocene-system-config-upgrade-multi-chain/README.md:21]:
```suggestion
This upgrades the `SystemConfig` in the
[op-contracts/v1.8.0-rc.3](https://github.com/ethereum-optimism/optimism/tree/op-contracts/v1.8.0-rc.3) release.
```

File: tasks/sep/020-holocene-system-config-upgrade-multi-chain/README.md
Comment (mds1) [inline tasks/sep/020-holocene-system-config-upgrade-multi-chain/README.md:25]: How can I verify this address is correct?

File: tasks/sep/020-holocene-system-config-upgrade-multi-chain/README.md
Comment (mds1) [inline tasks/sep/020-holocene-system-config-upgrade-multi-chain/README.md:30]:
```suggestion
When simulating, ensure the logs say `Using script /your/path/to/superchain-ops/tasks/sep/020-holocene-system-config-upgrade-multi-chain/NestedSignFromJson.s.sol`.
```

File: tasks/sep/020-holocene-system-config-upgrade-multi-chain/README.md
Comment (mds1) [inline tasks/sep/020-holocene-system-config-upgrade-multi-chain/README.md:40]: This sentence is incomplete. We can update this by pulling some text from the gov post

File: tasks/sep/020-holocene-system-config-upgrade-multi-chain/VALIDATION.md
Comment (mds1) [inline tasks/sep/020-holocene-system-config-upgrade-multi-chain/VALIDATION.md:1]: We are missing a few things in this doc:
- State overrides
- Nonce increments
- The approve hash calculation

See all my comments on the `VALIDATION.md` file in https://github.com/ethereum-optimism/superchain-ops/pull/374 for more info on how to easily fill out all of these

File: tasks/sep/020-holocene-system-config-upgrade-multi-chain/NestedSignFromJson.s.sol
Comment (mds1) [inline tasks/sep/020-holocene-system-config-upgrade-multi-chain/NestedSignFromJson.s.sol:31]: Same comment as https://github.com/ethereum-optimism/superchain-ops/pull/374#discussion_r1854420283

File: tasks/sep/020-holocene-system-config-upgrade-multi-chain/NestedSignFromJson.s.sol
Comment (mds1) [inline tasks/sep/020-holocene-system-config-upgrade-multi-chain/NestedSignFromJson.s.sol:38]: In the setup you can run some precheck assertions to make sure these env vars were set correctly, such as https://github.com/ethereum-optimism/superchain-ops/blob/c9a5f9b2ff8cfabd143ef5bc1eabdb0349632b27/tasks/eth/010-1-guardian-upgrade/NestedSignFromJson.s.sol#L85

File: tasks/sep/020-holocene-system-config-upgrade-multi-chain/NestedSignFromJson.s.sol
Comment (mds1) [inline tasks/sep/020-holocene-system-config-upgrade-multi-chain/NestedSignFromJson.s.sol:59]: Is this right, the semver should have a beta in it? I thought we got rid of the beta identifiers before deploying on Sepolia. cc @clabby

File: tasks/sep/020-holocene-system-config-upgrade-multi-chain/README.md
Comment (geoknee) [reply tasks/sep/020-holocene-system-config-upgrade-multi-chain/README.md:25]: Yes I just spotted this too, I think we need to investigate the tooling used to deploy that contract to understand the discrepancy.

Comment (geoknee) [reply tasks/sep/020-holocene-system-config-upgrade-multi-chain/README.md:25]: I used the address from the logs here https://github.com/ethereum-optimism/optimism/issues/13009#issuecomment-2493469236. Looks like @sebastianst specified this address when doing his ops, such that the `SystemConfig` was never deployed.

I can redeploy the correct version of this contract and then update this PR.

**Comment by geoknee** [comment]
> Fyi, this is the not task `19` but `20` because this one is currently an open PR [here](https://github.com/ethereum-optimism/superchain-ops/pull/374) and I assume would be merged before the current one.

I'm not sure what order they will end up being merged in. But #374 was definitely created first and reserves the task number 19.

I thought I had labelled everything here as task 20 -- did I miss something?

File: tasks/sep/020-holocene-system-config-upgrade-multi-chain/OVERVIEW.md
Comment (geoknee) [reply tasks/sep/020-holocene-system-config-upgrade-multi-chain/OVERVIEW.md]: [4e2aa1a](https://github.com/ethereum-optimism/superchain-ops/pull/378/commits/4e2aa1a29a585c90885615f70af48de6cd800278)

File: tasks/sep/020-holocene-system-config-upgrade-multi-chain/README.md
Comment (geoknee) [reply tasks/sep/020-holocene-system-config-upgrade-multi-chain/README.md]: [7e1de44](https://github.com/ethereum-optimism/superchain-ops/pull/378/commits/7e1de4404b72274448b7dbd80d4b13c63db21cf3)

File: tasks/sep/020-holocene-system-config-upgrade-multi-chain/README.md
Comment (geoknee) [reply tasks/sep/020-holocene-system-config-upgrade-multi-chain/README.md]: Updated here [f7080e0](https://github.com/ethereum-optimism/superchain-ops/pull/378/commits/f7080e0113dda3a00e984f01a822733db0fe6d9f). 

You can validate this on etherscan https://sepolia.etherscan.io/address/0x33b83E4C305c908B2Fc181dDa36e230213058d7d#code

File: tasks/sep/020-holocene-system-config-upgrade-multi-chain/NestedSignFromJson.s.sol
Comment (geoknee) [reply tasks/sep/020-holocene-system-config-upgrade-multi-chain/NestedSignFromJson.s.sol]: [f7080e0](https://github.com/ethereum-optimism/superchain-ops/pull/378/commits/f7080e0113dda3a00e984f01a822733db0fe6d9f)

**Comment by Ethnical** [comment]
> I thought I had labelled everything here as task 20 -- did I miss something?
That make sense to me! 
I prefer to write this in the comment if someone else have a look there and don't see the task 19 already merged in the repo. 
In the future, maybe we should inform this in the description of the current PR. 
This is a small nit but maybe worth it to remove unnecessary confusion in the future tasks :)

**Review by Ethnical — COMMENTED** [review]
Overall looks really nice, after all the fixes addressed I will approve this one.

File: tasks/sep/020-holocene-system-config-upgrade-multi-chain/NestedSignFromJson.s.sol
Comment (geoknee) [reply tasks/sep/020-holocene-system-config-upgrade-multi-chain/NestedSignFromJson.s.sol]: Do you mean as in here? https://github.com/ethereum-optimism/superchain-ops/blob/4bf34f1924ec3241674103d41fc076a54551edd5/tasks/sep/020-holocene-system-config-upgrade-multi-chain/NestedSignFromJson.s.sol#L110-L111

**Comment by geoknee** [comment]
@Ethnical should I add anything into CI to check this?

File: tasks/sep/020-holocene-system-config-upgrade-multi-chain/VALIDATION.md
Comment (geoknee) [reply tasks/sep/020-holocene-system-config-upgrade-multi-chain/VALIDATION.md:1]: [1c0d836](https://github.com/ethereum-optimism/superchain-ops/pull/378/commits/1c0d8361191d81adff967d9b41a947b2305070cf), [7bad130](https://github.com/ethereum-optimism/superchain-ops/pull/378/commits/7bad1303d6bd722ed54f0fd3904c90582f2886ee), [ef832c9](https://github.com/ethereum-optimism/superchain-ops/pull/378/commits/ef832c941cd992dd97866a766510e7fad2bde1e7) I copy pasted these from other tasks, please double check

File: tasks/sep/020-holocene-system-config-upgrade-multi-chain/NestedSignFromJson.s.sol
Comment (geoknee) [reply tasks/sep/020-holocene-system-config-upgrade-multi-chain/NestedSignFromJson.s.sol]: [19d9e21](https://github.com/ethereum-optimism/superchain-ops/pull/378/commits/19d9e21bee7a88c76e8fb76b4f230e3cc310d82f)

File: tasks/sep/020-holocene-system-config-upgrade-multi-chain/NestedSignFromJson.s.sol
Comment (geoknee) [reply tasks/sep/020-holocene-system-config-upgrade-multi-chain/NestedSignFromJson.s.sol]: [4413fb5](https://github.com/ethereum-optimism/superchain-ops/pull/378/commits/4413fb5f4ebec49179cc9f6c55410ece310cc85e)

File: tasks/sep/020-holocene-system-config-upgrade-multi-chain/.env
Comment (geoknee) [reply tasks/sep/020-holocene-system-config-upgrade-multi-chain/.env:4]: [e95ac21](https://github.com/ethereum-optimism/superchain-ops/pull/378/commits/e95ac21dfb78adf8bbdc9f61e2aeefafaa52b7d8)

File: tasks/sep/020-holocene-system-config-upgrade-multi-chain/NestedSignFromJson.s.sol
Comment (Ethnical) [reply tasks/sep/020-holocene-system-config-upgrade-multi-chain/NestedSignFromJson.s.sol]: @geoknee yes correct here :)

File: tasks/sep/020-holocene-system-config-upgrade-multi-chain/README.md
Comment (geoknee) [reply tasks/sep/020-holocene-system-config-upgrade-multi-chain/README.md]: [4c735c6](https://github.com/ethereum-optimism/superchain-ops/pull/378/commits/4c735c6d5f0bb5ac990fb021a89d0f03e276b346)

File: tasks/sep/020-holocene-system-config-upgrade-multi-chain/README.md
Comment (geoknee) [reply tasks/sep/020-holocene-system-config-upgrade-multi-chain/README.md]: [4c735c6](https://github.com/ethereum-optimism/superchain-ops/pull/378/commits/4c735c6d5f0bb5ac990fb021a89d0f03e276b346) Yeah that's right. I removed this entire sentence.

File: tasks/sep/020-holocene-system-config-upgrade-multi-chain/NestedSignFromJson.s.sol
Comment (geoknee) [reply tasks/sep/020-holocene-system-config-upgrade-multi-chain/NestedSignFromJson.s.sol]: [4c735c6](https://github.com/ethereum-optimism/superchain-ops/pull/378/commits/4c735c6d5f0bb5ac990fb021a89d0f03e276b346)

Comment (geoknee) [reply tasks/sep/020-holocene-system-config-upgrade-multi-chain/NestedSignFromJson.s.sol]: [4c735c6](https://github.com/ethereum-optimism/superchain-ops/pull/378/commits/4c735c6d5f0bb5ac990fb021a89d0f03e276b346)

File: tasks/sep/020-holocene-system-config-upgrade-multi-chain/VALIDATION.md
Comment (Ethnical) [inline tasks/sep/020-holocene-system-config-upgrade-multi-chain/VALIDATION.md:29]:
```suggestion
- [Etherscan (Council Safe)](https://sepolia.etherscan.io/address/0xf64bc17485f0B4Ea5F06A96514182FC4cB561977). This address is attested to the [Optimism docs](https://docs.optimism.io/chain/security/privileged-roles#l1-proxy-admin), as it's one of the signers of the L1 Proxy Admin owner.
```

File: tasks/sep/020-holocene-system-config-upgrade-multi-chain/VALIDATION.md
Comment (Ethnical) [inline tasks/sep/020-holocene-system-config-upgrade-multi-chain/VALIDATION.md:30]:
```suggestion
- [Etherscan (Foundation Safe)](https://sepolia.etherscan.io/address/0xDEe57160aAfCF04c34C887B5962D0a69676d3C8B). This address is attested to the [Optimism docs](https://docs.optimism.io/chain/security/privileged-roles#l1-proxy-admin), as it's one of the signers of the L1 Proxy Admin owner.
```

File: tasks/sep/020-holocene-system-config-upgrade-multi-chain/VALIDATION.md
Comment (Ethnical) [inline tasks/sep/020-holocene-system-config-upgrade-multi-chain/VALIDATION.md:146]:
```suggestion
- One on the ProxyAdminOwner (2/2) with address`0x1Eb2fFc903729a0F03966B917003800b145F56E2`.
```

File: tasks/sep/020-holocene-system-config-upgrade-multi-chain/VALIDATION.md
Comment (Ethnical) [inline tasks/sep/020-holocene-system-config-upgrade-multi-chain/VALIDATION.md:139]: I am not sure to understand this state overrides with the `livenessGuard`. 
I need to check this one before merging the PR, you got more infos about this change would be happy to have more infos.

File: tasks/sep/020-holocene-system-config-upgrade-multi-chain/README.md
Comment (Ethnical) [reply tasks/sep/020-holocene-system-config-upgrade-multi-chain/README.md]: Thanks nice I have double check this is the correct implementation now.

File: tasks/sep/020-holocene-system-config-upgrade-multi-chain/VALIDATION.md
Comment (Ethnical) [reply tasks/sep/020-holocene-system-config-upgrade-multi-chain/VALIDATION.md:1]: Looks good to me! 
Except the Liveness Guard that is currently incorrect.  I have proposed a change to fix this here: https://github.com/ethereum-optimism/superchain-ops/pull/378#discussion_r1862434630

File: tasks/sep/020-holocene-system-config-upgrade-multi-chain/VALIDATION.md
Comment (geoknee) [reply tasks/sep/020-holocene-system-config-upgrade-multi-chain/VALIDATION.md]: Ok done here [0726e80](https://github.com/ethereum-optimism/superchain-ops/pull/378/commits/0726e80299d23a023cfe0c20bc19f6a0adc86fdf). TBH I don't really understand this, I was just following instructions to copy paste from previous tasks.

File: tasks/sep/020-holocene-system-config-upgrade-multi-chain/.env
Comment (Ethnical) [reply tasks/sep/020-holocene-system-config-upgrade-multi-chain/.env:4]: Forgot to update the `.env` but looks nice:
```suggestion
# OWNER_SAFE = ProxyAdminOwner
OWNER_SAFE=0x1Eb2fFc903729a0F03966B917003800b145F56E2
```

**Comment by Ethnical** [comment]
Looks, Correct.
I have executed run the task and check the tenderly simulation:
image

The validation matches the current states changes.

**Review by Ethnical — APPROVED** [review]
Approve! 
I have added a small discussion for explanation on the livenessguard [here](https://github.com/ethereum-optimism/superchain-ops/pull/378#discussion_r1863329005)
Also, I have added one request to complete the sentence and then we can merge :)

File: tasks/sep/020-holocene-system-config-upgrade-multi-chain/VALIDATION.md
Comment (geoknee) [reply tasks/sep/020-holocene-system-config-upgrade-multi-chain/VALIDATION.md:1]: [0726e80](https://github.com/ethereum-optimism/superchain-ops/pull/378/commits/0726e80299d23a023cfe0c20bc19f6a0adc86fdf)

File: tasks/sep/020-holocene-system-config-upgrade-multi-chain/VALIDATION.md
Comment (geoknee) [reply tasks/sep/020-holocene-system-config-upgrade-multi-chain/VALIDATION.md]: [be28707](https://github.com/ethereum-optimism/superchain-ops/pull/378/commits/be28707b4abc854a677c5e53cf1d08debb8fef0f)

**Review by Ethnical — APPROVED** [review]

**Pull request merged**
